dags/analyzer_helper/discourse/extract_raw_info.py

```diff
@@ -0,0 +1,230 @@
+        WHERE id(post) = post_id
+        MATCH (topic:DiscourseTopic)-[:HAS_POST]->(post)
+        OPTIONAL MATCH (category:DiscourseCategory)-[:HAS_TOPIC]->(topic)
+        RETURN
+        id(post) AS post_id,
+        id(category) AS category_id,
+        category.name AS category_name
+        """
+        with self.driver.session() as session:
+            result = session.run(query, post_ids=post_ids)
+            records = [record.data() for record in result]
+            print(f"Cathegories fetched: ", records)
+            self.driver.close()
+            # print(f"Number of records fetched in fetch_post_categories: {len(records)}")
+            return records
+    
+    def get_latest_post_created_at(self, forum_endpoint: str) -> Optional[str]:
+        """
+        Fetches the created_at timestamp of the latest post from a specified forum.
+
+        Args:
+            uri (str): The URI to connect to the Neo4j database.
+            user (str): The username for the Neo4j database.
+            password (str): The password for the Neo4j database.
+            forum_endpoint (str): The endpoint of the Discourse forum.
+
+        Returns:
+            Optional[str]: The created_at timestamp of the latest post, or None if no posts are found.
+        """
+        
+        query = """
+        MATCH (forum:DiscourseForum {endpoint: $forum_endpoint})
+        WITH forum
+        MATCH (topic:DiscourseTopic {forumUuid: forum.uuid})
+        MATCH (topic)-[:HAS_POST]->(post:DiscoursePost)
+        RETURN post.createdAt AS created_at
+        ORDER BY post.createdAt DESC
+        LIMIT 1
+        """
+        
+        with self.driver.session() as session:
+            result = session.run(query, forum_endpoint=forum_endpoint)
+            latest_post = result.single()
+            
+            self.driver.close()
+        
+        if latest_post:
+            return latest_post["created_at"]
+        else:
+            return None
+        
+    def combine_posts_with_cathegories(self, post_details, post_categories) -> list:
+        """
+        Combine post details with their respective categories.
+
+        :param post_details: List of dictionaries containing post details.
+        :param post_categories: List of dictionaries containing post categories.
+        :return: List of combined dictionaries.
+        """
+        combined_results = []
+        for post in post_details:
+            matched = False
+            for category in post_categories:
+                if post["post_id"] == category["post_id"]:
+                    post["category_id"] = category["category_id"]
+                    post["category_name"] = category["category_name"]
+                    matched = True
+                    break
+            if not matched:
+                post["category_id"] = None
+                post["category_name"] = None
+            combined_results.append(post)
+        return combined_results
```

Comment:
This process is intensive lines 148 to 161. I asked GPT to simplify it and here's the results (you could also use your way to simplify it but please make sure it would work). 

```
def combine_post_details(post_details, post_categories):
    category_lookup = {category["post_id"]: category for category in post_categories}
    
    for post in post_details:
        category = category_lookup.get(post["post_id"])
        if category:
            post["category_id"] = category["category_id"]
            post["category_name"] = category["category_name"]
        else:
            post["category_id"] = None
            post["category_name"] = None
    
    return post_details
```